### ISSUE TITLE
Improve the screens when buying or selling items

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#51]
```
What steps will reproduce the problem?
1.
2.
3.

What is the expected output? What do you see instead?


What version of the product are you using? On what operating system?


Please provide any additional information below.

i find myself buying and selling items that i dont want to being that im using 
a touchscreen phone and the icons are quite small. an "are you sure" window 
would be nice
```

Original issue reported on code.google.com by `derrickv...@yahoo.com` on 3 Dec 2010 at 12:19

**Comment by GoogleCodeExporter** [comment]
```
im sorry that i didnt fill this out correctly. i noticed that after i posted it 
and couldnt delete it to make a new one. i have the 0.6.6 version on an android 
phone.
```

Original comment by `derrickv...@yahoo.com` on 3 Dec 2010 at 2:03

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report. No problem about the formatting.

This is actually a very good idea I think, and something that we should 
implement. It should be a setting though, so that players that want the old 
behaviour still can get that.

Additionally, such a popup dialog could let the player choose the amount of 
items that the transaction should affect. Let's say you want to buy 10 minor 
health potions, then the player could specify that in this dialog box.
```

Original comment by `oskar.wi...@gmail.com` on 3 Dec 2010 at 8:46

**Comment by GoogleCodeExporter** [comment]
```
longclick on sell could pop up such a dialog. Options could be:
- oldschool (as it is now)
- always ask amount/confirm
- longclick brings up dialog 
- oldschool but longclick buys/sells all

dialog could contain:
- amount 
- buttons: all (includes "yes" or simply writes max to amount); yes; cancel
```

Original comment by `surrano` on 8 Dec 2010 at 12:32

**Comment by GoogleCodeExporter** [comment]
```
I suggest we re-order the buy/sell screens.
- remove [Info] button
- add longclick option to [sell] buton
- add longclick option to [[item icon][item text/title]]
- move "Your gold: X" to the far left
- add new gold column into content area, so it becomes 4 ([item icon][item 
text/title][action btton][gold])
- add "Gold [gold icon]" to the far right on the row as "Your gold" is on. This 
should fit the width of column 3 & 4
- rename "Buy (X gold)" and "Sell (X gold)" to simply "Buy" and "Sell" 
respectfully
- item buy/sell values are presented in the new fourth gold column by numeric 
value only

Single/short clicks on item row does nothing like now.

Item longclik options:
- Info about the item
- No buy/sell button, only close

Buy/Sell button longclick options:
- Item count field with +/- icons. Holding +/- increases the count field value 
and the longer you hold the faster it goes up/down. Minimum and default value 
is 1 (avoids 0 items buy/sell actions)
- Item max button. For buy, maximum in stock and for sell, maximum you have
- Buy/sell and close buttons

The longclick options could be combined under item row, but it feels more right 
to have them separated.

This should clear up the screens, prevent accidental buys/sells where you 
wanted to click [Info] and eliminate the need for a confirm dialog box, but it 
could be optional for buy/sell actions where item count is above 1, 10 or some 
other (custom?) number.
```

Original comment by `kims...@gmail.com` on 19 Dec 2010 at 11:19

**Comment by GoogleCodeExporter** [comment]
```
I fail to understand how "[sell] X gold" is better than "[sell (X gold)]". 
Approx. same total space required, but the sell button is smaller and easier to 
miss in small screens. Longclick on name for info and longclick on buy/sell 
could be the same, as long as no tedious info about any single item (i.e. info 
+ amount manipulators + sell + close all fit the same dialog)
```

Original comment by `surrano` on 21 Dec 2010 at 7:13

**Comment by GoogleCodeExporter** [comment]
Original comment by `oskar.wi...@gmail.com` on 10 Feb 2011 at 7:05
- Changed title: **Improve the screens when buying or selling items**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
ready for 0.6.10
```

Original comment by `SamuelPl...@gmail.com` on 21 Jul 2011 at 8:37

**Comment by GoogleCodeExporter** [comment]
Original comment by `oskar.wi...@gmail.com` on 23 Jul 2011 at 12:10
- Changed state: **Ready**
- Added labels: **Milestone-0.6.10**

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.10 . Closing.
```

Original comment by `oskar.wi...@gmail.com` on 21 Oct 2011 at 8:56
- Changed state: **Fixed**
